lib/index.js

```diff
@@ -66,48 +157,22 @@ module.exports = {
       files: ['jquery.js'],
     });
 
-    let emberSourceDistPath = path.join(__dirname, '..', 'dist');
-
-    let emberCliBabel = this.addons.find(a => a.name === 'ember-cli-babel');
-    let needsLegacyBuild = [
-      'transform-literals',
-      'transform-arrow-functions',
-      'transform-destructuring',
-      'transform-spread',
-      'transform-parameters',
-      'transform-computed-properties',
-      'transform-shorthand-properties',
-      'transform-block-scoping',
-      'transform-classes',
-    ].some(p => emberCliBabel.isPluginRequired(p));
-
-    if (needsLegacyBuild) {
-      emberSourceDistPath = path.join(emberSourceDistPath, 'legacy');
-    }
+    let ember;
+    let targets = (this.project && this.project.targets && this.project.targets.browsers) || [];
 
-    let emberFiles = [
-      'ember-template-compiler.js',
-      'ember-testing.js',
-      'ember.debug.js',
-      'ember.min.js',
-      'ember.prod.js',
-    ]
-      .map(function(file) {
-        return [file, file.replace('.js', '.map')];
-      })
-      .reduce(function(flat, jsAndMap) {
-        return flat.concat(jsAndMap);
-      }, [])
-      .filter(function(file) {
-        let fullPath = path.join(emberSourceDistPath, file);
-
-        return fs.existsSync(fullPath);
+    if (!isProduction && PRE_BUILT_TARGETS.every(target => targets.includes(target))) {
+      ember = new Funnel(tree, {
+        destDir: 'ember',
+        include: ['ember.debug.js', 'ember.debug.map', 'ember-testing.js', 'ember-testing.map'],
```

Comment:
Should `ember-template-compiler.js` be present here as well?

This change might have caused this: https://github.com/embroider-build/embroider/issues/311.

Reply:
That seems plausible, mind making a PR so we can focus discussion over there?

Reply:
That's done at https://github.com/emberjs/ember.js/pull/18291.